# ONA config

topics:
  input:
    lidar: /ona2/sensors/pandar_front/cloud
    imu: /ona2/sensors/imu_front/imu
  # output:

num_threads: 20 # OpenMP threads (will be equal to $(nproc) if it is set higher)

sensor_type: 2  # LiDAR type (0: OUSTER \ 1: VELODYNE \ 2: HESAI \ 3: LIVOX)

debug: true     # save useful intermediate pcl (pc2match, deskewed...)
verbose: true   # print debug/performance board

estimate_extrinsics: true   # continuous estimation of LiDAR-IMU extrinsics 
time_offset: true           # whether to take into account the offset between sweep reference time and first point timestamp

calibration:    # automatic IMU calibration
  gravity_align: true
  accel: true
  gyro: true
  time: 3.0

extrinsics:
  baselink2imu:
    t: [ 0.87, 0.44, 0.17 ]
    R: [ 1.,  0.,  0.,
          0.,  1.,  0.,
          0.,  0.,  1. ]
  baselink2lidar:
    t: [ 0.395,  -0.885,  0.201 ]
    R: [ 0.774, -0.633, 0.022,
        0.633, 0.774,  -0.026,
        -0.0005, 0.03,  0.999 ]

intrinsics:
  accel:
    bias: [ 0.0, 0.0, 0.0 ]
    sm:   [ 1.,  0.,  0.,
            0.,  1.,  0.,
            0.,  0.,  1. ]
  gyro:
    bias: [ 0.0, 0.0, 0.0 ]

filters:
  cropBox:
    active: true
    box:
      min: [ -1.0, -1.0, -1.0 ]
      max: [ 1.0, 1.0, 1.0 ]

  voxelGrid:
    active: true
    leafSize: [ 0.25, 0.25, 0.25 ]

iKFoM:
  MAX_NUM_ITERS: 3
  NUM_MATCH_POINTS: 5
  MAX_DIST_PLANE: 2.0
  PLANES_THRESHOLD: 5.0e-2
  LIMITS: 0.001
  covariance:
    gyro: 0.0001 
    accel: 0.01
    bias_gyro: 0.00001
    bias_accel: 0.0001
